conda/recipes/libcudf/meta.yaml

```diff
@@ -29,16 +29,18 @@ requirements:
   host:
     - librmm {{ minor_version }}.*
     - cudatoolkit {{ cuda_version }}.*
-    - arrow-cpp 0.15.0.*
+    - arrow-cpp 0.17.1.*
+    - arrow-cpp-proc * cuda
     - double-conversion
     - rapidjson
     - flatbuffers
     - boost-cpp
     - dlpack
   run:
     - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}
-    - arrow-cpp 0.15.0.*
     - dlpack
+  run_constrained:
+    - arrow-cpp-proc * cuda
```

Comment:
@jakirkham somehow it seems like this + using a run_export from `arrow-cpp` isn't playing nicely with libcudf being installed where it's allowing an `arrow-cpp` `_cpu` build when building `cudf`. Any ideas here? Unless we need to add `arrow-cpp-proc * cuda` to the `host` section of `cudf` recipe as well?

Reply:
Yeah adding it to `host` and `run` seems ok to me.

Idk if there is a good way to get `run_exports` to pick up the build string and not the version. Will look into it though. Can see how that would help simplify things for us.